crates/autopilot/src/run_loop.rs

```diff
@@ -622,19 +755,25 @@ impl RunLoop {
     async fn settle(
         &self,
         driver: &infra::Driver,
-        solved: &competition::SolutionWithId,
+        solution_id: u64,
+        solved_order_uids: HashSet<OrderUid>,
         auction_id: i64,
         submission_deadline_latest_block: u64,
     ) -> Result<(), SettleError> {
         let request = settle::Request {
-            solution_id: solved.id(),
+            solution_id,
             submission_deadline_latest_block,
         };
         let tx_hash = self
             .wait_for_settlement(driver, auction_id, request)
             .await?;
         tracing::debug!(?tx_hash, "solution settled");
 
+        self.in_flight_orders
```

Comment:
This is not required since the same set added is removed later. The only issue was that the cache wouldn't have been cleaned up if the settlement had failed. I fixed that. Thanks!